src/main/java/uk/co/bconline/ndelius/service/impl/UserServiceImpl.java

```diff
@@ -109,11 +115,19 @@ public UserServiceImpl(
 		val dbFuture = supplyAsync(() -> userEntityService.search(query, includeInactiveUsers, datasetFilter), taskExecutor);
 		val ldapFuture = supplyAsync(() -> userEntryService.search(query, includeInactiveUsers, datasetFilter), taskExecutor);
 
+		val roleFuture = supplyAsync(() -> roleService.getUsersRoles(role).stream()
```

Comment:
Are we invoking the role search every time? Or are we able to skip this processing if the `role` isn't provided?